pkg/util/clusteragent/clcrunner.go

```diff
@@ -0,0 +1,151 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+package clusteragent
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/api/security"
+	"github.com/DataDog/datadog-agent/pkg/api/util"
+	"github.com/DataDog/datadog-agent/pkg/clusteragent/clusterchecks/types"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/version"
+)
+
+/*
+Client to query the Datadog Cluster Level Check Runner API.
+*/
+
+const (
+	clcRunnerPath = "api/v1/clcrunner"
+)
+
+var globalCLCRunnerClient *CLCRunnerClient
+
+// CLCRunnerClientInterface is required to query the API of Datadog Cluster Level Check Runner
+type CLCRunnerClientInterface interface {
+	GetVersion(IP string) (version.Version, error)
+	GetRunnerStats(IP string) (types.CLCRunnersStats, error)
+}
+
+// CLCRunnerClient is required to query the API of Datadog Cluster Level Check Runner
+type CLCRunnerClient struct {
+	clcRunnerAPIRequestHeaders http.Header
+	clcRunnerAPIClient         *http.Client
+	clcRunnerPort              int
+}
+
+// resetGlobalCLCRunnerClient is a helper to remove the current CLCRunnerClient global
+// It is ONLY to be used for tests
+func resetGlobalCLCRunnerClient() {
+	globalClusterAgentClient = nil
+}
+
+// GetCLCRunnerClient returns or init the CLCRunnerClient
+func GetCLCRunnerClient() (CLCRunnerClientInterface, error) {
+	if globalCLCRunnerClient == nil {
+		globalCLCRunnerClient = &CLCRunnerClient{}
+		if err := globalCLCRunnerClient.init(); err != nil {
+			log.Debugf("CLC Runner client init error: %v", err)
+			return nil, err
+		}
+	}
+	return globalCLCRunnerClient, nil
+}
+
+func (c *CLCRunnerClient) init() error {
+	authToken, err := security.GetClusterAgentAuthToken()
+	if err != nil {
+		return err
+	}
+
+	// Set headers
+	c.clcRunnerAPIRequestHeaders = http.Header{}
+	c.clcRunnerAPIRequestHeaders.Set(authorizationHeaderKey, fmt.Sprintf("Bearer %s", authToken))
+
+	// Set http client
+	// TODO remove insecure
+	c.clcRunnerAPIClient = util.GetClient(false)
+	c.clcRunnerAPIClient.Timeout = 2 * time.Second
+
+	// Set http port used by the CLC Runners
+	c.clcRunnerPort = config.Datadog.GetInt("cluster_checks.clc_runners_port")
+
+	return nil
+}
+
+// GetVersion fetches the version of the CLC Runner
+func (c *CLCRunnerClient) GetVersion(IP string) (version.Version, error) {
+	const clcRunnerVersionPath = "version"
```

Comment:
worth moving with the other const at the top ?

Reply:
tried to keep the same style we use in the DCA client in the same package

https://github.com/DataDog/datadog-agent/blob/master/pkg/util/clusteragent/clusteragent.go#L141-L143

https://github.com/DataDog/datadog-agent/blob/master/pkg/util/clusteragent/clusteragent.go#L200-L201

https://github.com/DataDog/datadog-agent/blob/master/pkg/util/clusteragent/clusteragent.go#L235-L236